habitat-baselines/habitat_baselines/rl/multi_agent/pop_play_wrappers.py

```diff
@@ -120,14 +127,24 @@ def get_extra(
         return _merge_list_dict(all_extra)
 
     @classmethod
-    def from_config(cls, config, observation_space, action_space, **kwargs):
-        return MultiPolicy()
+    def from_config(
+        cls,
+        config,
+        observation_space,
+        action_space,
+        update_obs_with_agent_prefix_fn: Optional[Callable] = None,
+        **kwargs,
+    ):
+        return MultiPolicy(update_obs_with_agent_prefix_fn)
 
 
 class MultiStorage(Storage):
-    def __init__(self):
+    def __init__(self, update_obs_with_agent_prefix_fn):
         self._active_storages = []
         self._agent_type_ids = []
+        if update_obs_with_agent_prefix_fn is None:
```

Comment:
Why do you need this now? Before we always had the agent prefix and didn't define this function. What changed?

Reply:
I need this so I can inject more information into the observation.

Reply:
Extra information is the behavior latent